mparray/special/_special.py

```diff
@@ -109,20 +121,30 @@ def expit(x):  # needs accuracy review
     return mp.exp(x - mp.log1p(mp.exp(x)))
 
 
-@vectorize
-def boxcox(x, lmbda):  # needs accuracy review
+def _boxcox_scalar(x, lmbda):
     """
     y = (x**lmbda - 1) / lmbda  if lmbda != 0
         log(x)                  if lmbda == 0
     """
+    if x < 0:
+        return mp.nan
     if lmbda != 0:
         return mp.powm1(x, lmbda) / lmbda
     else:
-        return mp.lmbda(x)
+        return mp.log(x)
+
+
+@vectorize
+def boxcox(x, lmbda):
+    return _boxcox_scalar(x, lmbda)
 
 
-def boxcox1p(x, lmbda):  # needs accuracy review
-    return boxcox(mp.one + x, lmbda)
+def boxcox1p(x, lmbda):
```

Comment:
Doesn't this need to be vectorized, then?

Reply:
Good catch on this and below. What happened is that I copied this over from a branch where I implemented the inverses of the boxcox functions. There I also have a `_boxcox1p_scalar` which is used internally by `inv_boxcox1p`. This is just the scalar version which is why it has the unnecessary check and lacks vectorization.

Reply:
OK. Will you add it, or do you want it merged as-is?

Reply:
I’ll add the inverses in a future PR. This boxcox1p needs to be fixed before merging though.